node_tree.py

```diff
@@ -322,6 +322,11 @@ def init(self, context):
                 sys.stderr.write('ERROR: %s\n' % str(err))
             self.set_color()
 
+    def sv_new_input(self, socket_type, name, attrib_dict):
```

Comment:
```suggestion
    def sv_new_input(self, socket_type, name, **attrib_dict):
```
As to me it would be a little bit more flexible. 
It will be still possible to use dictionary `self.sv_new_input(**{'attribute': value})`
and in the same time you have option to pass attributes as normal parameters `self.sv_new_input(attribute=value)`